sql/driver/wire.proto

```diff
@@ -87,6 +71,36 @@ message Request {
 }
 
 message Response {
+  // Result corresponds to the execution of a single SQL statement.
+  message Result {
+    // DDL is an empty message that is used in Result's oneof. Its
+    // presence indicates that the operation was a DDL statement.
+    message DDL {}
+
+    // Rows encapsulates the result of an operation that returns rows.
+    message Rows {
+      // A Row is a collection of values representing a row in a result.
+      message Row {
+        repeated Datum values = 1 [(gogoproto.nullable) = false];
+      }
+      // The names of the columns returned in the result set in the order
+      // specified in the SQL statement. The number of columns will equal
+      // the number of values in each Row.
+      repeated string columns = 1;
+      // The rows in the result set.
+      repeated Row rows = 2 [(gogoproto.nullable) = false];
+    }
+
+    // Error is non-nil if an error occurred while executing the statement.
+    optional string error = 1;
+
+    oneof union {
+      DDL ddl = 2 [(gogoproto.customname) = "DDL"];
+      uint32 rows_affected = 3;
```

Comment:
Cool, thanks for the feedback. I'm going to proceed with this as-is. @petermattis I chose `rows_affected` because that's the terminology used by Go's sql package.